src/AmalgamatorSession.ts

```diff
@@ -675,6 +696,30 @@ export class AmalgamatorSession extends LoggingDebugSession {
             ].customRequest('cdt-gdb-adapter/Memory', args);
             response.body = childResponse.body;
             this.sendResponse(response);
+        } else if (command === 'cdt-amalgamator/resumeAll') {
+            const [, threads] = await this.collectChildTheads();
+            await Promise.all(
+                threads.map(async (thread) => {
+                    const [childIndex] = await this.getThreadInfo(thread.id);
+                    const childResponse = await this.childDaps[
+                        childIndex
+                    ].threadsRequest();
+                    const threadInfo = childResponse.body
+                        .threads[0] as ThreadInfo;
+                    if (threadInfo.running === false) {
+                        const continueResponse =
+                            response as DebugProtocol.ContinueResponse;
+                        const continueArgs = {
+                            threadId: thread.id,
+                        } as DebugProtocol.ContinueArguments;
+                        this.continueRequest(continueResponse, continueArgs);
```

Comment:
This does the wrong thing, you can see the bad behaviour in the log. 

```
[14:13:19.889 UTC] From client: cdt-amalgamator/resumeAll(undefined)
[14:13:50.417 UTC] To client: {"seq":0,"type":"event","event":"continued","body":{"threadId":1000}}
[14:18:29.551 UTC] To client: {"seq":0,"type":"event","event":"continued","body":{"threadId":1001}}
[14:18:36.189 UTC] To client: {"seq":0,"type":"response","request_seq":16,"command":"cdt-amalgamator/resumeAll","success":true}
[14:18:36.190 UTC] To client: {"seq":131,"type":"response","request_seq":16,"command":"cdt-amalgamator/resumeAll","success":true,"body":{"allThreadsContinued":false}}
[14:18:36.191 UTC] To client: {"seq":131,"type":"response","request_seq":16,"command":"cdt-amalgamator/resumeAll","success":true,"body":{"allThreadsContinued":false}}
```

In the above log you can see a few issues:

1. The continued event is sent before the child has finished its continue. This is a problem because if the child fails to continue, the continued event will already have been sent
2. The amalgamator is sending "extra" response packets to the client because of the reuse of continueRequest. In the log you can see that for my two child system, the amalgamator is sending 3 resumeAll responses.

As far a a fix, I think you need the following:

1. await the long running operations. You do a await `Promise.all`, but your threads.map does not return any non-null promises to wait for
2. factor out continueRequest, or better still just replace this line with `await childDap.continueRequest(args);` as a few lines above you have already determined the thread id of the child dap.


_PS. Sorry for not catching the above error yesterday, I incorrectly thought it had something to do with the now removed code._

Reply:
> 2\. factor out continueRequest, or better still just replace this line with `await childDap.continueRequest(args);` as a few lines above you have already determined the thread id of the child dap.

I updated it.

Reply:
> * await the long running operations. You do a await `Promise.all`, but your threads.map does not return any non-null promises to wait for
> * factor out continueRequest, or better still just replace this line with `await childDap.continueRequest(args);` as a few lines above you have already determined the thread id of the child dap.

I have updated it. When I look at the log file, the order sent is correct. But I'm not sure if it's correct or not.
![Screenshot_51](https://user-images.githubusercontent.com/104537989/218836665-20f70364-9fa0-4669-9e18-e38c0aa09c09.png)

Could you please help me review it?

Reply:
I can see from your log that the `continued` event has the wrong thread id. It seems to have the thread id of the childdap instead of the one that VScode client expects.

This is what I see:

```
[20:04:10.498 UTC] From client: cdt-amalgamator/resumeAll(undefined)
[20:04:10.504 UTC] To client: {"seq":0,"type":"event","event":"continued","body":{"threadId":1}}
[20:04:10.505 UTC] To client: {"seq":0,"type":"event","event":"continued","body":{"threadId":1}}
[20:04:10.505 UTC] To client: {"seq":0,"type":"response","request_seq":16,"command":"cdt-amalgamator/resumeAll","success":true}
```

However, despite the `threadId` being incorrect, VSCode still shows the threads transitioning from suspended to running. Therefore I realized that the continued events are not being processed as expected. This is because we need to send `allThreadsContinued` field. The `allThreadsContinued` has three possible states, `undefined`, `true`, `false` (the type is `?:boolean`, i.e. an optional boolean). So we need to explicitly say either `allThreadsContinued` `false` or perhaps we can send a single `allThreadsContinued` as `true`, although my guess is the code logic on the former is easier.

Reply:
I updated the `allThreadsContinued` field for the event and also changed the `thread id` as required.